Plugins/SpatialGDK/SpatialGDKEditorToolbar/Source/SpatialGDKEditorToolbar/Private/InteropCodeGenerator/TypeBindingGenerator.cpp

```diff
@@ -149,6 +151,90 @@ FString PropertyToWorkerSDKType(UProperty* Property, bool bIsRPCProperty)
 	return DataType;
 }
 
+// Generate definition in header file for Blueprint-defined Structs & Enums
+void GenerateBlueprintStructEnumDefinition(FCodeWriter& Writer, UProperty* Property, TSet<FString>& GeneratedTypes, TMap<FString, FString>& GeneratedStructInfo)
+{
+	if (GeneratedTypes.Contains(Property->GetCPPType()))
+	{
+		return;
+	}
+
+	if (UArrayProperty *ArrProp = Cast<UArrayProperty>(Property))
+	{
+		GenerateBlueprintStructEnumDefinition(Writer, ArrProp->Inner, GeneratedTypes, GeneratedStructInfo);
+	}
+
+	if (Property->IsA<UStructProperty>())
+	{
+		UStruct* Struct = Cast<UStructProperty>(Property)->Struct;
+		if (!Struct->IsA<UUserDefinedStruct>())
+		{
+			return;
+		}
+
+		GeneratedTypes.Add(Property->GetCPPType());
+		GeneratedStructInfo.Add(Property->GetCPPType(), Struct->GetPathName());
+
+		// Recurse children post-order
+		for (TFieldIterator<UProperty> PropIt(Struct); PropIt; ++PropIt)
+		{
+			UProperty* Prop = *PropIt;
+			GenerateBlueprintStructEnumDefinition(Writer, Prop, GeneratedTypes, GeneratedStructInfo);
+		}
+
+		Writer.PrintNewLine();
+		Writer.Printf("struct %s", *Property->GetCPPType());
+		Writer.Printf("{").Indent();
+		for (TFieldIterator<UProperty> PropIt(Struct); PropIt; ++PropIt)
+		{
+			UProperty* Prop = *PropIt;
+			FStringOutputDevice PropertyText;
+			Prop->ExportCppDeclaration(PropertyText, EExportedDeclaration::Local, nullptr);
+			Writer.Printf("%s;", *PropertyText);
+		}
+		Writer.Outdent().Printf("};");
+		Writer.Printf("UPROPERTY()");  // To-do: remove this line once the UHT bugfix goes in: case sensitive matching USTRUCT
```

Comment:
What is this bugfix? When is it going in?

Reply:
Making PR right now -- was discussing with Valentyn

Reply:
https://github.com/improbableio/UnrealEngine/pull/6